pkg/diff/diff.go

```diff
@@ -207,16 +213,12 @@ func applyDefaultValues(result *typed.TypedValue) ([]byte, error) {
 	return mergedBytes, nil
 }
 
-func buildDiffResult(mergedBytes []byte, live *unstructured.Unstructured) (*DiffResult, error) {
+func buildDiffResult(predictedBytes []byte, liveBytes []byte) (*DiffResult, error) {
 
-	liveBytes, err := json.Marshal(live)
-	if err != nil {
-		return nil, fmt.Errorf("error while marshaling live unstructured: %w", err)
-	}
 	return &DiffResult{
-		Modified:       string(liveBytes) != string(mergedBytes),
+		Modified:       string(liveBytes) != string(predictedBytes),
```

Comment:
Is the string conversion necessary for this comparison?

Reply:
get the error:
`operator != not defined for []byte`